Tests/SwiftDocCTests/Semantics/MarkupReferenceResolverTests.swift

```diff
@@ -27,4 +27,11 @@ class MarkupReferenceResolverTests: XCTestCase {
         _ = resolver.visit(document)
         XCTAssertEqual(0, resolver.problems.count)
     }
+
+    func testDuplicatedDiagnosticForExtensionFile() throws {
+        let (_, context) = try testBundleAndContext(named: "ExtensionArticleBundle")
+        // Before #733's change, there will be 2 duplicated problems.
+        // And we'll trigger a index out of range crash in DefaultDiagnosticConsoleFormatter
+        XCTAssertEqual(1, context.problems.count)
+    }
```

Comment:
Should we also verify that this problem comes from the expected source file?
```suggestion
        XCTAssertEqual("Server.md", context.problems.first?.diagnostic.source?.lastPathComponent)
    }
```